src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -102,23 +102,34 @@
  * &lt;module name=&quot;TrailingComment&quot;/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * // OK
+ * if (&#47;&#42; OK &#42;&#47; x &#62; 5) {}
+ * int a = 5; // violation
+ * doSomething(
+ *   param1
+ * ); // OK, by default such trailing of method/code-block ending is allowed
+ * </pre>
+ * <p>
  * To configure the check to enforce only comment on a line:
  * </p>
  * <pre>
  * &lt;module name=&quot;TrailingComment&quot;&gt;
- *   &lt;property name=&quot;format&quot; value=&quot;^\\s*$&quot;/&gt;
+ *   &lt;property name=&quot;format&quot; value=&quot;^\s*$&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
- * Example for trailing comments check to suppress specific trailing comment:
+ * Example:
  * </p>
  * <pre>
- * public class Test {
- *   int a; // SUPPRESS CHECKSTYLE
- *   int b; // NOPMD
- *   int c; // NOSONAR
- *   int d; // violation, not suppressed
- * }
+ * // OK
+ * if (&#47;&#42; OK, this comment does not end the line &#42;&#47; x &#62; 5) {}
+ * int a = 5; // violation, string before comment should match pattern "^\s*$"
```

Comment:
`// violation, line content before comment should match pattern "^\s*$"`

apply to other line like this.

Reply:
Done!

Reply:
Done!